src/lib/math/pcurves/pcurves_impl/pcurves_wrap.h

```diff
@@ -17,6 +17,11 @@ concept curve_supports_fe_invert2 = requires(const typename C::FieldElement& fe)
    { C::fe_invert2(fe) } -> std::same_as<typename C::FieldElement>;
 };
 
+template <typename C>
+concept curve_supports_scalar_invert = requires(const typename C::FieldElement& fe) {
+   { C::scalar_invert(fe) } -> std::same_as<typename C::FieldElement>;
+};
```

Comment:
```suggestion
template <typename C>
concept curve_supports_scalar_invert = requires(const typename C::Scalar& s) {
   { C::scalar_invert(s) } -> std::same_as<typename C::Scalar>;
};
```

Copy&Paste issue? Note that this would just silently disable your optimized code. There won't be a compiler error; the concept just wouldn't fit.

Reply:
Eep this would explain why I was suddenly struggling to repro the improvements on my other machine ... thank you :sweat_smile:

Reply:
It's all internal APIs, but perhaps a smoke test would be a good idea. 😏 I tested the concepts locally with `std::cout` in the respective branches.